javaee/impl/src/main/java/org/jboss/forge/addon/javaee/jpa/ui/JPANewEntityCommandImpl.java

```diff
@@ -86,13 +105,43 @@ public JavaClassSource decorateSource(UIExecutionContext context, Project projec
       {
          idStrategyChosen = GenerationType.AUTO;
       }
+
+      EntityIdType idTypeChosen = idType.getValue();
+      if(idTypeChosen == null){
+          idTypeChosen = EntityIdType.LONG_PROPERTY;
+      }
+
       ConfigurationFacet facet = project.getFacet(ConfigurationFacet.class);
       Configuration config = facet.getConfiguration();
       String idPropertyName = config.getString(PersistenceOperations.ID_PROPERTY_NAME_CONFIGURATION_KEY,
                configuration.getString(PersistenceOperations.ID_PROPERTY_NAME_CONFIGURATION_KEY, "id"));
       String versionPropertyName = config.getString(PersistenceOperations.VERSION_PROPERTY_NAME_CONFIGURATION_KEY,
                configuration.getString(PersistenceOperations.VERSION_PROPERTY_NAME_CONFIGURATION_KEY, "version"));
-      return persistenceOperations.newEntity(source, idStrategyChosen, tableName.getValue(), idPropertyName,
-               versionPropertyName);
+
+      JavaResource jr = project.getFacet(JavaSourceFacet.class).getJavaResource(idClass.getValue());
+
+      if(!(jr.getJavaType() instanceof JavaClassSource)){
+          throw new IllegalArgumentException("The found JavaResource for the ID class is not a JavaClassResource, " +
+                  "which is required. Found resource: " + jr.getClass().getCanonicalName());
+      }
+
+      JavaClassSource jcs = (JavaClassSource) jr.getJavaType();
```

Comment:
Uhm, this breaks if `LONG_PROPERTY` should be used... :D

Reply:
I'll put it into an own function that is called in the `ID_CLASS` and `EMBEDDED_ID` case below.